src/System.Private.CoreLib/src/System/Buffer.cs

```diff
@@ -117,10 +117,40 @@ public static int ByteLength(Array array)
             return _ByteLength(array);
         }
 
-        internal static unsafe void ZeroMemory(byte* src, long len)
+        // This is currently used by System.IO.UnmanagedMemoryStream
+        internal static unsafe void ZeroMemory(byte* dest, long len)
         {
-            while (len-- > 0)
-                *(src + len) = 0;
+#if BIT64
+            ZeroMemory(dest, (ulong)(len));
+#else
+            ulong length = (ulong)(len);
```

Comment:
This is not necessary. The buffer length is always going to fit into 32-bits on 32-bit platforms. This can be just:

```
Debug.Assert(len == (nuint)len);
ZeroMemory(dest, (nuint)len);
```

Reply:
Fixed.